packages/web/src/components/RedwoodApolloProvider.tsx

```diff
@@ -20,12 +23,46 @@ const ApolloProviderWithFetchConfig: React.FunctionComponent<{
   config?: Omit<ApolloClientOptions<InMemoryCache>, 'cache'>
 }> = ({ config = {}, children }) => {
   const { uri, headers } = useFetchConfig()
+  const { getToken, type: authProviderType, isAuthenticated } = useAuth()
+
+  const withToken = setContext(async () => {
+    if (isAuthenticated && getToken) {
+      const token = await getToken()
+
+      return { token }
+    }
+
+    return { token: null }
+  })
+
+  const authMiddleware = new ApolloLink((operation, forward) => {
+    const { token } = operation.getContext()
+
+    // Only add auth headers when token is present
+    // Token is null, when !isAuthenticated
+    const authHeaders = token
+      ? {
+          'auth-provider': authProviderType,
+          authorization: token ? `Bearer ${token}` : null,
```

Comment:
This check seems a bit redundant.
```suggestion
          authorization: `Bearer ${token}`
```

Reply:
Without this check, graphql calls fail when you're not logged in.

Reply:
Are you sure? It looks like it wouldn't hit the `null` branch here since you're doing the same check on line 43-44.

Reply:
Oh sorry you're right!